### PR TITLE
OpenMC mat fix

### PR DESCRIPTION
**Pull request by pshriwise**
This PR updates OpenMC material writing in the following ways:

  - no longer expands elements, but writes them in the GND format
  - ignores compositions with zero weight/atom fractions in the material composition
 
It adds tests which: 

  - ensure that reading/writing of materials is consistent for the OpenMC format
  - verify the correct output for compositions containing natural elements

### CODE REVIEW

**Review by paulromano — COMMENTED** [review]
OpenMC does need you to expand elements before writing XML. The only cases where it accepts elemental data is if the nuclear data library itself uses elemental data. The most common example is carbon which has historically been given as an elemental evaluation (although evaluations for C12 and C13 have now become available in ENDF/B-VIII.0). Other than carbon, everything else should probably be split out into nuclides. The way we handle this in OpenMC's Python API is to look at the user's `OPENMC_CROSS_SECTIONS` environment variable to figure out what data they actually have available. Not sure you'd really want to have that kind of logic here though...

**Review by gonuke — APPROVED** [review]
Looks OK to me, but I don't know anything about OpenMC formats

**Comment by pshriwise** [comment]
Welp. That was rather silly of me. I ran into that carbon issue when converting a DAGMC model and generalized it without looking more closely. 

I'll include a corner case for carbon for now.

**Comment by pshriwise** [comment]
I've updated the writer and added the option of passing a set of nucids to ignore in `expand_elements`. Added a test for that as well.

File: pyne/material.pyx [hunk 1]
```diff
@@ -680,10 +680,12 @@ cdef class _Material:
         """
         return self.mat_pointer.molecular_mass(atoms_per_molecule)
 
-    def expand_elements(self):
+    def expand_elements(self, nucset = set()):
```
Comment (paulromano) [inline pyne/material.pyx]: Need a `Parameters` section in the docstring below this

File: pyne/material.pyx [hunk 2]
```diff
@@ -680,10 +680,18 @@ cdef class _Material:
...
-        Exapnds the elements ('U', 'C', etc) in the material by replacing them
-        with their natural isotopic distributions.  This function returns a copy.
+        Exapnds the elements ('U', 'C', etc) in the material by
+        replacing them with their natural isotopic distributions with
+        the exception of the ids in nucset. This function returns a
+        copy.
+
+        Parameters
+        ----------
+	nucset : set, optional
```
Comment (paulromano) [inline pyne/material.pyx]: Indentation looks like it got messed up here (and below as well)

File: pyne/material.pyx [hunk 3]
```diff
@@ -680,10 +680,18 @@ cdef class _Material:
         """
         return self.mat_pointer.molecular_mass(atoms_per_molecule)
 
-    def expand_elements(self):
+    def expand_elements(self, nucset=set()):
         """expand_elements(self)
-        Exapnds the elements ('U', 'C', etc) in the material by replacing them
-        with their natural isotopic distributions.  This function returns a copy.
+        Exapnds the elements ('U', 'C', etc) in the material by
```
Comment (paulromano) [inline pyne/material.pyx]: Typo: Exapnds

**Review by paulromano — APPROVED** [review]
Looks good to me now!

**Comment by pshriwise** [comment]
Any further comments @gonuke?

**Comment by gonuke** [comment]
Already approved...
